test/zdtm/static/memfd-secret00.c

```diff
@@ -0,0 +1,80 @@
+#include <stdio.h>
+#include <string.h>
+#include <sys/mman.h>
+#include <sys/syscall.h>
+#include <unistd.h>
+
+#include "zdtmtst.h"
+
+#define SECRET "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+#define SIZE   26
+
+const char *test_doc = "memfd_secret file descriptor";
+const char *test_author = "Dhanuka Warusadura <csx@tuta.io>";
+
+static int _memfd_secret(unsigned int flags)
+{
+#ifdef __NR_memfd_secret
+	return syscall(__NR_memfd_secret, flags);
+#else
+	return -1;
+#endif
```

Comment:
Understood. But, how can I gracefully skip the test? There is no [API](https://criu.org/ZDTM_API) to skip the test.  I think the feature check (.desc file) approach is the only way to skip the test.

Reply:
You can take a look at test/zdtm/static/arm-neon00.c and test/zdtm/static/fpu02.c to see how they skip tests based on an `#ifdef`. 

For the run-time, you can check if `syscall(__NR_memfd_secret)` returned `-ENOSYS` and then just print message with `skip()` and exit with success.

Reply:
Can we just define __NR_memfd_secret? (E.g. like we define __NR_rseq. For all arches except alpha it should be 447, and we don't support arch/alpha as far as I remember.)

Reply:
@rppt included the conditional compilation and call to `skip()`.

Reply:
@Snorch thanks. But, I'm not sure it will work. Because, as I understand, even if we define the ` __NR_memfd_secret`, the `secretmem` feature needs to be enable to use the syscall. I think, without explicitly enabling  `secretmem`, exposing/defining `__NR_memfd_secret` has no use.

Reply:
@warusadura Let me explain:
1) You need a kerndat check which will test if `secretmem` feature is supported by current kernel. And based on the result of this check you can skip the test as we do in multiple other tests.
2) You need to define __NR_memfd_secret as local headers may not provide it even if the current kernel supports it. I imagine that your ifdef is trying to do similar thing, just defining is much more clear solution.
Those are two separate things.

Reply:
Understood, thanks you @Snorch

Reply:
implemented,
```
#ifndef __NR_memfd_secret
#define __NR_memfd_secret 447
#endif
```